src/commands/dev/gcs/server.rs

```diff
@@ -49,6 +49,24 @@ pub(super) async fn serve(
                 // what it contains and display in logs
                 let (parts, body) = req.into_parts();
 
+                if parts.method == Method::OPTIONS && server_config.allowed_origins.len() > 0 {
+                    return async move {
+                        let response_builder = Response::builder()
+                            .header("Access-Control-Allow-Credentials", "true")
+                            .header(
+                                "Access-Control-Allow-Methods",
+                                "GET, HEAD, POST, PUT, DELETE, OPTIONS",
+                            )
+                            .header("Access-Control-Allow-Origin", "http://localhost:3000")
+                            .header("Access-Control-Allow-Headers", "Content-Type");
+                        for allowed_origin in server_config.allowed_origins {
+                            response_builder
+                                .header("Access-Control-Allow-Origin", allowed_origin.to_string());
+                        }
+                        Ok::<_, failure::Error>(response_builder.body(()))
+                    };
+                }
```

Comment:
this part super doesnt compile right now but this is sorta what im thinking for handling OPTIONS requests when `--cors` args are passed